pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/BookkeeperSchemaStorage.java

```diff
@@ -126,33 +128,54 @@ public CompletableFuture<StoredSchema> get(String key, SchemaVersion version) {
 
     @Override
     public CompletableFuture<List<CompletableFuture<StoredSchema>>> getAll(String key) {
+        return getAll(key, null);
+    }
+
+    public CompletableFuture<List<CompletableFuture<StoredSchema>>> getAll(String key,
+            LocatorEntry locator) {
         CompletableFuture<List<CompletableFuture<StoredSchema>>> result = new CompletableFuture<>();
-        getLocator(key).thenAccept(locator -> {
-            if (log.isDebugEnabled()) {
-                log.debug("[{}] Get all schemas - locator: {}", key, locator);
-            }
 
-            if (!locator.isPresent()) {
-                result.complete(Collections.emptyList());
-                return;
-            }
+        if (log.isDebugEnabled()) {
+            log.debug("[{}] Get all schemas - locator: {}", key, locator);
+        }
+        List<CompletableFuture<StoredSchema>> list = new ArrayList<>();
+        if (locator == null) {
```

Comment:
We can just add 

```java
CompletableFuture<Optional<LocatorEntry>> locatorFuture = 
                locator == null ? getLocator(key) : CompletableFuture.completedFuture(Optional.of(locator));
locatorFuture.thenAccept(...)
```
to reduce the duplicated code.

Reply:
fixed